pandas/tests/extension/test_categorical.py

```diff
@@ -207,6 +207,20 @@ def test_cast_nan_to_int(self, cls, values):
         with pytest.raises((ValueError, TypeError), match=msg):
             s.astype(int)
 
+    @pytest.mark.parametrize(
```

Comment:
can you add the test for the original issue as well (leave this here it is good), that test the merging.

Reply:
Added a test for the merging in either direction since that was relevant to the issue, but unfortunately the dtype of the column depends on join order (is this still a bug?) so I had to force the dtypes to be equal before the assertion.

As a side note I tried setting `check_dtype=False` in `assert_frame_equal` and got this error, which seems odd since I'm asking it to ignore dtypes: `AssertionError: IntervalArray Expected type <class 'pandas.core.arrays.interval.IntervalArray'>, found <class 'pandas.core.arrays.numpy_.PandasArray'> instead`